public/app/plugins/datasource/elasticsearch/configuration/ElasticDetails.tsx

```diff
@@ -108,22 +108,13 @@ export const ElasticDetails = ({ value, onChange }: Props) => {
   );
 };
 
-// TODO: Use change handlers from @grafana/data
-const changeHandler =
-  (key: keyof DataSourceSettings<ElasticsearchOptions>, value: Props['value'], onChange: Props['onChange']) =>
-  (event: React.SyntheticEvent<HTMLInputElement | HTMLSelectElement>) => {
-    onChange({
-      ...value,
-      [key]: event.currentTarget.value,
-    });
-  };
-
 // TODO: Use change handlers from @grafana/data
 const jsonDataChangeHandler =
   (key: keyof ElasticsearchOptions, value: Props['value'], onChange: Props['onChange']) =>
   (event: React.SyntheticEvent<HTMLInputElement | HTMLSelectElement>) => {
     onChange({
       ...value,
+      database: key === 'index' ? '' : value.database,
```

Comment:
i understand that this approach will work, but i find it to be a surprising behavior. by that i mean, when quickly looking through this file, and seeing the function with the name `jsonDataChangeHandler`, i would not assume that it will do something special for some special fields.

may i recommend a different approach?

duplicate the function `jsonDataChangeHandler` under a new name, for example `indexChangeHandler`, and make it do the thing needed for the index-change, and call it from the index field's `onChange`. you may even be able to simplify the `indexChangeHandler` case.

yes, it will duplicate code a little, but i like that more than the other way.

what do you think?